sound/drivers/xen-front.c

```diff
@@ -161,6 +161,111 @@ static struct sdev_pcm_stream_info *sdrv_stream_get(
 	return stream;
 }
 
+static inline void sdrv_alsa_timer_rearm(struct sdev_alsa_timer_info *dpcm)
+{
+	mod_timer(&dpcm->timer, jiffies +
+		(dpcm->frac_period_rest + dpcm->rate - 1) / dpcm->rate);
+}
+
+static void sdrv_alsa_timer_update(struct sdev_alsa_timer_info *dpcm)
+{
+	unsigned long delta;
+
+	delta = jiffies - dpcm->base_time;
+	if (!delta)
+		return;
+	dpcm->base_time += delta;
+	delta *= dpcm->rate;
```

Comment:
what about integer overflow there? Remember, that `jiffies` will overflow  eventually. So, you'll get very big value in `delta`. This is not problem, until you use right comparison functions (you are not using them, BTW). But you can't just multiply `delta` with something and expect meaningful result.